src/bdmsupport.c

```diff
@@ -743,10 +794,35 @@ int bdmUpdateDeviceData(item_list_t *itemList)
             pDeviceData->bdmDeviceType = BDM_TYPE_ILINK;
         else if (!strcmp(pDeviceData->bdmDriver, "sdc") && strlen(pDeviceData->bdmDriver) == 3)
             pDeviceData->bdmDeviceType = BDM_TYPE_SDC;
-        else
+        else if (!strcmp(pDeviceData->bdmDriver, "ata") && strlen(pDeviceData->bdmDriver) == 3) {
+            pDeviceData->bdmDeviceType = BDM_TYPE_ATA;
+            itemList->flags = MODE_FLAG_COMPAT_DMA;
+        } else
             pDeviceData->bdmDeviceType = BDM_TYPE_UNKNOWN;
 
-        LOG("Mass device: %d (%d) %s -> %s\n", itemList->mode, pDeviceData->massDeviceIndex, pDeviceData->bdmPrefix, pDeviceData->bdmDriver);
+        // If the device is backed by the ATA driver then get the supported LBA size for the drive.
+        if (pDeviceData->bdmDeviceType == BDM_TYPE_ATA) {
+            // If atad is loaded then xhdd is also loaded, query the hdd to see if it supports LBA48 or not.
+            pDeviceData->bdmHddIsLBA48 = fileXioDevctl("xhdd0:", ATA_DEVCTL_IS_48BIT, NULL, 0, NULL, 0);
+            if (pDeviceData->bdmHddIsLBA48 < 0) {
+                // Failed to query the LBA limit of the device, fail safe to LBA28.
+                LOG("Mass device %d is backed by ATA but failed to get LBA limit %d\n", pDeviceData->massDeviceIndex, pDeviceData->bdmHddIsLBA48);
+                pDeviceData->bdmHddIsLBA48 = 0;
+            }
+            // Query the drive for the highest UDMA mode.
```

Comment:
#ifdef ATA_UDMA_PLUS should be moved here, if we still use that.

Reply:
should I just remove the ifdef entirely since we're always going to be using it?

Reply:
I've gone ahead and just removed it, if you're good with that I'll rebase into the first commit